### PR TITLE
Improve doc for pallet macro and config macro

### PR DESCRIPTION
**Pull request by thiolliere**
Improve doc:

* the pallet macro is actually referring to 2 places, for the module and for the struct placeholder but doesn't really clarify it (I should have named the latter just `pallet_struct` or something but it is a bit late)

* The doc of `with_default` is a bit confusing too IMO.

CC @kianenigma

### CODE REVIEW

File: substrate/frame/support/src/lib.rs [hunk 1]
```diff
@@ -936,6 +936,79 @@ pub mod pallet_prelude {
...
+/// Note that the `dev_mode` argument can only be supplied to the `#[pallet]` or
+/// `#[frame_support::pallet]` attribute macro that encloses your pallet module. This
+/// argument cannot be specified anywhere else, including but not limited to the
+/// `#[pallet::pallet]` attribute macro.
+///
+/// <div class="example-wrap" style="display:inline-block"><pre class="compile_fail"
+/// style="white-space:normal;font:inherit;">
+/// <strong>WARNING</strong>:
+/// You should not deploy or use dev mode pallets in production. Doing so can break your
+/// chain and therefore should never be done. Once you are done tinkering, you should
```
Comment (liamaharon) [inline substrate/frame/support/src/lib.rs:1005]:
```suggestion
/// You should never deploy or use dev mode pallets in production. Doing so can break your
/// chain. Once you are done tinkering, you should
```

File: substrate/frame/support/src/lib.rs [hunk 2]
```diff
@@ -936,6 +936,79 @@ pub mod pallet_prelude {
...
+///   these will be replaced by `Blake2_128Concat`. In case of explicit key-binding, `Hasher`
+///   can simply be ignored when in `dev_mode`.
+///
+/// Note that the `dev_mode` argument can only be supplied to the `#[pallet]` or
+/// `#[frame_support::pallet]` attribute macro that encloses your pallet module. This
+/// argument cannot be specified anywhere else, including but not limited to the
+/// `#[pallet::pallet]` attribute macro.
+///
+/// <div class="example-wrap" style="display:inline-block"><pre class="compile_fail"
+/// style="white-space:normal;font:inherit;">
```
Comment (liamaharon) [inline substrate/frame/support/src/lib.rs:1002]: Could you please post a screenshot of how this looks?

Comment (thiolliere) [reply substrate/frame/support/src/lib.rs:1002]: ok, but this is not my code this is cut/paste from below. I only added the syntax part


Capture d’écran du 2024-04-16 14-44-56

Comment (thiolliere) [reply substrate/frame/support/src/lib.rs:1002]: Capture d’écran du 2024-04-16 14-47-14

**Review by liamaharon — APPROVED** [review]

File: substrate/frame/support/src/lib.rs [hunk 3]
```diff
@@ -936,6 +936,83 @@ pub mod pallet_prelude {
...
+///
+/// 	#[pallet::call]
+/// 	impl<T: Config> Pallet<T> {
+/// 	}
+///
+/// 	/* ... */
+/// }
+/// ```
+///
+/// The documentation for each individual part can be found at [frame_support::pallet_macros]
```
Comment (franciscoaguirre) [inline substrate/frame/support/src/lib.rs:965]: Does this result in a link?

**Review by franciscoaguirre — APPROVED** [review]

File: substrate/frame/support/src/lib.rs (hunk 3 above)
Comment (thiolliere) [reply substrate/frame/support/src/lib.rs:965]: Yes, I think rust doc link directly to item when there are in scope.

**Review by gupnik — APPROVED** [review]
Thank you!

**Pull request merged**